crates/hir-ty/src/infer/unify.rs

```diff
@@ -438,6 +438,7 @@ impl<'a> InferenceTable<'a> {
     where
         T: HasInterner<Interner = Interner> + TypeFoldable<Interner>,
     {
+        // TODO check this vec here
```

Comment:
what sort of check would happen here?

Reply:
Oh that was a note to myself to check if we can cache the allocation for that vec here